cmd/nerdctl/run.go

```diff
@@ -693,7 +694,11 @@ func generateRootfsOpts(ctx context.Context, client *containerd.Client, platform
 		}
 		var processArgs []string
 		if entrypoint != "" {
-			processArgs = append(processArgs, entrypoint)
+			s, err := shellwords.Parse(entrypoint)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+			processArgs = append(processArgs, s...)
```

Comment:
Seems incompatible with Docker.
Docker doesn't parse the entrypoint.
```
$ docker run --rm --entrypoint "echo foo bar" alpine
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "echo foo bar": executable file not found in $PATH: unknown.
```

Reply:
I think we should just change the type of the `entrypoint` from `string` to `[]string`.

Reply:
@AkihiroSuda I am rethinking about it , switching `entrypoint` to slice this means sthg like this 
`docker run --rm  --entrypoint "foo" --entrypoint "echo"  alpine` which is confusing.
FYI `docker-compose --entrypoint` is parsing shell words.
`nerdctl compose` rely on `nerdctl` which is not the same behavior for docker-compose and docker. Keeping 100% compatibility  is tough in this situation

Reply:
> docker run --rm --entrypoint "foo" --entrypoint "echo" alpine which is confusing.

I think this is fine as an internal implementation detail of `nerdctl compose`.
`nerdctl run` users are not expected to specify multiple `--entrypoint` flags manually.

Reply:
fixed